urlce7759b8da_Palachsuper_EBA4_tgz-pJ8-json_TestJ8/src/terminal/service/Service.java

```diff
@@ -0,0 +1,193 @@
+           str= cmd.charAt(k);
+           aaa.append(str);
+        }
+        cmd = aaa+"";
+        WritePaymentLog(cmd, cashe_machine.id_payment);
+        String result = cmd;
+         try
+         {
+             result = java.net.URLDecoder.decode(result, "UTF-8");
+             JSONObject json_cmd = new JSONObject(result);
+             String start_cmd = json_cmd.getString("command");
+             try {
+                 MyEnum e = MyEnum.valueOf(start_cmd);
+                 switch (e) {
+                     case print:
+                         printer.main(json_cmd);  // Отправили на в функцию печать JSON
+                         break;
+                     case dispense:
+                         dispencer.main(cmd);
+                         break;
+                     case card:
+                         card_reader.main(cmd);
+                         break;
+                     case cashe:
+                         cashe_machine.main(json_cmd);
+                         break;
+                 }
+             } catch (IllegalArgumentException iae) {
+
+             }
+             String pr_string = json_cmd.getString("params");
+             JSONObject param = new JSONObject(pr_string);
+         }
+         catch (Exception e){
+
+         }
+    }
+
+    public static void WritePaymentLog(String log, String id_payment){
+        StringBuilder sb = new StringBuilder();
+        try{
+            String path_log = "/home/terminal/Log/LogPayment/Payment_"+id_payment+".txt";
+            File Log_path = new File(path_log);
+            if (!Log_path.exists()){
+                try {
+                    Log_path.createNewFile();
+                } catch (IOException e) {
+                    // TODO Auto-generated catch block
+                    e.printStackTrace();
+                }
+            }
+            BufferedReader in = new BufferedReader(new FileReader( new File(path_log).getAbsoluteFile()));
+            String read_log;
+            if (sb.length()>0)
+            {
+                sb.delete(0,sb.length()-1);
+            }
+            while ((read_log = in.readLine()) != null)
+            {
+                if (sb.length()>0)
+                {
+                    sb.append(read_log);
+                    sb.append("\n");
+                }
+            }
+            in.close();
+            String str="";
+            str =sb + log;
+            PrintStream printStream = new PrintStream(new FileOutputStream(path_log, true), true);
+            printStream.println(str);
+            printStream.close();
+        }
+        catch (Exception e){
+
+        }
+    }
+
+    public static void WriteLog(String log){
+        String dat_time_path="";
+        StringBuilder sb = new StringBuilder();
+        try
+        {
+            try
+            {
+                Calendar cal = new GregorianCalendar();
+                String name_cal=new SimpleDateFormat("yyyyMMdd").format(cal.getTime());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.